core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +39,30 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store individually via {@link #write(MarshalledEntry)}.
+    * As the execution order of MarshalledEntries is not guaranteed, you should ensure that only a single entry exists
+    * in the Collection for a given key.
+    *
+    * @param entries a Collection of MarshalledEntry to be written to the store.
+    * @throws NullPointerException if entries is null.
+    */
+   default void writeBatch(Collection<MarshalledEntry<? extends K, ? extends V>> entries) {
```

Comment:
The consumer doesn't say that it says give me the next batch. To be honest I never meant to imply that `OffHeapDataContainer.entrySet` did it's own batching (I thought it would do 1 element at a time as well). Just kinda fell down the rabbit hole :)

But I am glad I looked into it, because `flatMap` is really a collecting intermediate operation, creates "batches", and I was unaware of that behavior. I had assumed it would lazily process each entry of the inner stream, but it first has to collect all entries of that stream. But knowing this I changed the `entrySet` so that it "batches" it per address pointer and if they ever change the behavior, we will be fine as well.

It seems the spec only specifies the source spliterator is the only that has to be lazy (unless a stateful operator is used).